tempesta_fw/http.c

```diff
@@ -3688,26 +3895,78 @@ tfw_http_hdr_split(TfwStr *hdr, TfwStr *name_out, TfwStr *val_out)
 		}
 	}
 
+	/* The header value is empty. */
+	if (unlikely(!val_found))
+		return;
+
+	if (WARN_ON_ONCE(!last_chunk))
+		return;
+
 	T_DBG3("%s: hdr_tail=%lu, val_out->len=%lu, last_tail=%lu,"
 	       " last_chunk->len=%lu, last_chunk->data='%.*s'\n", __func__,
 	       hdr_tail, val_out->len, last_tail, last_chunk->len,
 	       (int)last_chunk->len, last_chunk->data);
 
-	if (WARN_ON_ONCE(!last_chunk))
-		return -EINVAL;
-
 	val_out->nchunks = chunk - val_out->chunks;
 	val_out->len -= hdr_tail;
 	last_chunk->len -= last_tail;
+}
+
+/*
+ * Split header in two parts: name and value, with descriptors copying (see
+ * description of @tfw_http_hdr_split() for details).
+ *
+ * TODO: this procedure should be evicted after HTTP/1.1-parser extending
+ * (and @tfw_http_hdr_split()/@tfw_h2_msg_hdr_length procedures upgrading) in
+ * order to split headers into separate chunks for name, ':', LWS, value, and
+ * RWS - during the HTTP-message parsing stage.
```

Comment:
Just leave a comment to remember that the TODO must be fixed in upcoming PR.

Reply:
TODO is fixed.